src/main/scala/nl/knaw/dans/easy/multideposit/actions/AddBagToDeposit.scala

```diff
@@ -73,3 +80,30 @@ object AddBagToDeposit {
     }
   }
 }
+
+class BagInfoCompleter(bagFactory: BagFactory, dataset: Dataset) extends Completer {
+
+  def complete(bag: Bag): Bag = {
+    val newBag = bagFactory.createBag(bag)
+
+    // copy files from bag to newBag
+    newBag.putBagFiles(bag.getPayload)
+    newBag.putBagFiles(bag.getTags)
+
+    // create a BagInfoTxt based on the old one
+    val bagPartFactory = bagFactory.getBagPartFactory
+    val bagInfo = bagPartFactory.createBagInfoTxt(bag.getBagInfoTxt)
+
+    // add the CREATED field
+    dataset.get("DDM_CREATED")
+      .flatMap(_.find(s => !s.isBlank))
+      .map(DateTime.parse)
```

Comment:
Good point! The exception would have been caught eventually in the `createBag` function, but that is a little late to find that the date isn't formatted properly. Instead I added an extra check to `AddDatasetMetadataToDeposit` to validate that the date is formatted properly.